db/models.js

```diff
@@ -0,0 +1,109 @@
+const client = require('./index');
+const helpers = require('./helpers');
+
+const getReviews = (productId, sort, offset, count) => {
+  let sortValue;
+  if (sort === 'newest') {
+    sortValue = 'date DESC';
+  } else if (sort === 'helpful') {
+    sortValue = 'helpfulness DESC';
+  } else if (sort === 'relevant') {
+    sortValue = 'date DESC, helpfulness DESC';
+  } else { console.log('invalid sort value'); }
+
+  const query = {
+    text: `SELECT * FROM reviews WHERE product_id = $1 AND reported = false ORDER BY ${sortValue} OFFSET $2 LIMIT $3`,
+    values: [productId, offset, count],
+  };
+
+  return client.query(query)
+    .then((res) => res.rows)
+    .catch((error) => error.stack);
+};
+
+const getMeta = (productId) => {
+  const characteristicsQuery = `SELECT value, characteristics.id, name
+  FROM reviews
+  INNER JOIN characteristics_reviews
+  ON reviews.id = review_id AND reviews.product_id = ${productId}
+  INNER JOIN characteristics
+  ON characteristics.id = characteristic_id AND
+  characteristics.product_id = ${productId}`;
+
+  const ratingsRecommendQuery = `SELECT rating, recommend FROM reviews WHERE product_id = ${productId}`;
+
+  return Promise.all([
+    client.query(ratingsRecommendQuery),
+    client.query(characteristicsQuery),
+  ])
+    .then((res) => {
+      const ratingsRecommended = helpers.createRatingsRecommendedObj(res[0].rows);
+      const characteristics = helpers.createCharacteristicsObj(res[1].rows);
+      return [ratingsRecommended[0], ratingsRecommended[1], characteristics];
+    })
+    .catch((error) => error.stack);
+};
+
+const updateReviewsTable = (reviewDetails) => {
+  const updateReviewsQuery = 'INSERT INTO reviews(product_id, rating, date, summary, body, recommend, reported, reviewer_name, reviewer_email, response, helpfulness) VALUES ($1, $2, current_timestamp, $3, $4, $5, false, $6, $7, null, 0) RETURNING id';
+
+  const updateReviewsValues = [reviewDetails.product_id, reviewDetails.rating,
+    reviewDetails.summary, reviewDetails.body, reviewDetails.recommend,
+    reviewDetails.name, reviewDetails.email];
+
+  return client.query(updateReviewsQuery, updateReviewsValues)
+    .then((res) => res.rows[0].id)
+    .catch((error) => error.stack);
+};
+
+const updatePhotosTable = (reviewId, photosArray) => {
+  const updatePhotosQuery = helpers.createUpdatePhotosQuery(reviewId, photosArray);
+  return client.query(updatePhotosQuery)
+    .then((res) => console.log('Inserted rows:', res.rowCount))
+    .catch((error) => error.stack);
+};
+
+const updateCharacteristicsReviewsTable = (reviewId, characteristicsObject) => {
+  // eslint-disable-next-line max-len
+  const updateCharacteristicsReviewsQuery = helpers.createUpdateCharacteristicsReviewsQuery(reviewId, characteristicsObject);
+  return client.query(updateCharacteristicsReviewsQuery)
+    .then((res) => console.log('Inserted rows:', res.rowCount))
+    .catch((error) => error.stack);
+};
+
+async function postReview(reviewDetails) {
```

Comment:
@jabrake : After looking at your code, I just realized I also used async await in my db module, and I also forgot to use the try/catch block.

Reply:
Thanks for this one too, updated!